src/harness/testcases/WINNF_FT_S_DRG_testcase.py

```diff
@@ -269,8 +272,161 @@ def test_WINNF_FT_S_DRG_5(self):
+    heartbeat_c = {'operationState': 'GRANTED'}
+    heartbeat_b = {'operationState': 'GRANTED'}
+
+    # Deregistration requests (where testcode will fill in cbsdId)
+    deregister_a = {}
+    # device_c's CBSD ID will be removed due to the 'REMOVE' keyword.
+    deregister_c = {'cbsdId': 'REMOVE'}
+    deregister_b = {}
+
+    # Device_a and device_c are Category A.
+    self.assertEqual(device_a['cbsdCategory'], 'A')
+    self.assertEqual(device_c['cbsdCategory'], 'A')
+
+    # Device_b is Category B with conditionals pre-loaded.
+    self.assertEqual(device_b['cbsdCategory'], 'B')
+    conditionals_b = {
+        'cbsdCategory': device_b['cbsdCategory'],
+        'fccId': device_b['fccId'],
+        'cbsdSerialNumber': device_b['cbsdSerialNumber'],
+        'airInterface': device_b['airInterface'],
+        'installationParam': device_b['installationParam']
+    }
+    conditionals = {'registrationData': [conditionals_b]}
+    del device_b['installationParam']
+
+    # Create the actual config.
+    devices = [device_a, device_c, device_b]
+    grant_requests = [grant_a, grant_c, grant_b]
+    heartbeat_requests = [heartbeat_a, heartbeat_c, heartbeat_b]
+    deregistration_requests = [deregister_a, deregister_c, deregister_b]
+    config = {
+        'registrationRequest': devices,
+        'conditionalRegistrationData': conditionals,
+        'grantRequest': grant_requests,
+        'heartbeatRequest': heartbeat_requests,
+        'deregistrationRequest': deregistration_requests,
+        # First request, all valid params => SUCCESS
+        # Second request is missing CBSD ID => MISSING_PARAM
+        # Third request, all valid params => SUCCESS
+        'expectedResponseCodes': [(0,), (102,), (0,)]
+    }
+    writeConfig(filename, config)
+
+  @configurable_testcase(generate_DRG_6_default_config)
+  def test_WINNF_FT_S_DRG_6(self, config_filename):
+    """[Configurable] Array Deregistration."""
+
+    config = loadConfig(config_filename)
+    # Very light checking of the config file.
+    self.assertTrue(
+        len(config['registrationRequest']) == len(config['grantRequest']) ==
+        len(config['heartbeatRequest']) == len(config['deregistrationRequest']))
+    self.assertEqual(
+        len(config['deregistrationRequest']),
+        len(config['expectedResponseCodes']))
+
+    # Whitelist FCC IDs.
+    for device in config['registrationRequest']:
+      self._sas_admin.InjectFccId({
+          'fccId': device['fccId'],
+          'fccMaxEirp': 47
+      })
+
+    # Whitelist user IDs.
+    for device in config['registrationRequest']:
+      self._sas_admin.InjectUserId({'userId': device['userId']})
+
+    # Step 2 & 3: Register devices and get grants
+    cbsd_ids, grant_ids = self.assertRegisteredAndGranted(
+        config['registrationRequest'], config['grantRequest'],
+        config['conditionalRegistrationData'])
+
+    # Step 4: First Heartbeat Request
+    heartbeat_request = config['heartbeatRequest']
```

Comment:
As this step (step4) expects success for all heartbeat requests, the function assertHeartbeatsSuccessful can be used.

Reply:
While I agree that this is a good place to use that helper function, I slightly prefer the way it is now since its much clearer as to what the requests really are. (Also because we resend the same request in Step 8, according to the spec.) So, keeping it as is.

Reply:
Ok. I agree.